pymeasure/instruments/nf/li5660.py

```diff
@@ -0,0 +1,445 @@
+        "FREQ:SMUL?", ":FREQ:HARM ON; :FREQ:MULT %d",
+        """ A property that control the subharmonic order m for measurement (Primary PSD) """,
+        validator=truncated_range,
+        values=[1, 63]
+    )
+
+    secondary_frequency = Instrument.measurement(
+        ":FREQ2?",
+        """ Return the secondary frequency used with detection modes DUAL2 and CASCADE. """
+    )
+
+    harmonics_sec_enable = Instrument.control(
+        ":FREQ2:HARM?",":FREQ2:HARM %d",
+        """ A property that control the harmonic measurement (enabled or disabled) (secondary PSD)""",
+        validator=strict_discrete_set,
+        values={"On": 1, "Off": 0},
+        map_values=True
+    )
+
+    harmonics_sec_order = Instrument.control(
+        ":FREQ2:MULT?", ":FREQ2:HARM ON; :FREQ:MULT %d",
+        """ A property that control the harmonic order n for measurement (secondary PSD) """,
+        validator=truncated_range,
+        values=[1, 63]
+    )
+
+    phase_shift = Instrument.control(
+        ":PHAS?", ":PHAS %g",
+        """ A property that control the phase shift amount (primary PSD). """,
+        validator=truncated_range,
+        values=[-180.000, 179.000]
+    )
+
+    ##########
+    # Source #
+    ##########
+    source_frequency = Instrument.control(
+        ":SOUR:FREQ?", "SOUR:FREQ %d",
+        """ A property that control the internal oscillator (primary PSD) frequency. """,
+        validator=truncated_range,
+        values=[300e-3, 1.15e+7]
+    )
+
+    source_voltage = Instrument.control(
+        ":SOUR:VOLT?", ":SOUR:VOLT %d",
+        """ A property that control the internal oscillator output voltage AC amplitude. """,
+        validator=truncated_range,
+        values=[0, 1]
+    )
+
+    ############
+    # Methods  #
+    ############
+    def default_display_screen(self):
+        """ Configures the measurement screen. """
+        self.display_screen = "Fine"
+
+    def default_dynamic_reserve(self):
+        """ Configure the dynamic reserve Sensitivity. """
+        self.dynamic_reserve = "Medium"
+
+    def scrn(self, DISPLAY=None):
+        """ A property allows user to switch between the 'Standard', 'Expanded', 'Detail' screens. """
+        log.info("Screen is display in {}".format(self.name))
+        
+        if DISPLAY is None:
+            self.default_display_screen()
+        else:
+            self.display_screen = DISPLAY
+
+    def dr(self, DR=None):
+        """ A property allow user to set the sensitivity of the dynamic reserve. """
+        log.info("The dynamic reserve is set to {}".format(self.name))
+        
+        if DR is None:
+            self.default_dynamic_reserve()
+        else:
+            self.dynamic_reserve = DR
+
+    def slope(self, SLOPE=6):
+        """ A property allow user to set the filter attenuation slope (primary PSD). """
+        log.info("The filter attenuation slope is set to {}".format(self.name))
+
+        if SLOPE==6:
```

Comment:
The value-checking is not required over here I think; that is already done by the `self.filter_slope` property, right?

Reply:
Yes, you are right. "if else" statement removed.